src/safepoints.rs

```diff
@@ -0,0 +1,138 @@
+extern crate ykstackmaps;
+
+use std::collections::HashMap;
+use std::path::Path;
+use ykstackmaps::{LocKind, LocOffset, SMRec, StackMapParser};
+
+use core::mem;
+
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
+pub struct ReturnAddress(pub u64);
+
+/// The offset from the stack pointer.
+#[derive(Debug)]
+struct SPO(u32);
+
+/// A `PtrSlot` identifies a stack root at a given safepoint using its offset
+/// from the Stack Pointer.
+///
+/// A base pointer (not to be confused with X86 terminology, where base pointer
+/// refers to the frame pointer register) is a pointer an object. In opposition
+/// to this, a derived pointer points to the interior of an object.
+///
+/// The Derived variant of a `PtrSlot` also contains a Stack Pointer offset to
+/// the base of the object.
+#[derive(Debug)]
+enum PtrSlot {
+    Base(SPO),
+    Derived(SPO, SPO),
+}
+
+/// Contains root locations for a Safepoint
```

Comment:
Nitpick. All your other comments have a full stop on the end.

Reply:
Okay I'll fix this for consistency